### PR TITLE
Cherry-Pick Managed Identity Metadata fix for patch release 1.8.1

### PR DESCRIPTION
**Pull request by schaabs**

### CODE REVIEW

File: sdk/identity/Azure.Identity/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Release History
 
+## 1.8.1 (2023-01-13)
+
+### Bugs Fixed
+- Fixed an issue when using `ManagedIdentityCredential` in combination with authorities other than Azure public cloud that resulted in a bogus instance metadata validation error. [#32498](https://github.com/Azure/azure-sdk-for-net/issues/32498)
```
Comment (heaths) [inline sdk/identity/Azure.Identity/CHANGELOG.md:6]: Not sure who well "bogus" is understood or could be translated (though, I don't think we're doing that right now).
```suggestion
- Fixed an issue when using `ManagedIdentityCredential` in combination with authorities other than Azure public cloud that resulted in a incorrect metadata validation error. [#32498](https://github.com/Azure/azure-sdk-for-net/issues/32498)
```

Or maybe "an erroneous".

File: sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs [hunk 2]
```diff
@@ -100,6 +101,73 @@ public async Task VerifyImdsRequestWithClientIdMockAsync()
             Assert.AreEqual("true", metadataValue);
         }
 
+        [NonParallelizable]
+        [Test]
```
Comment (heaths) [inline sdk/identity/Azure.Identity/tests/ManagedIdentityCredentialTests.cs:105]: Tip: when using `[TestCase]` or similar attributes, you don't *need* `[Test]`. Doesn't hurt, though. Just a tip.

**Review by christothes — APPROVED** [review]

**Review by heaths — APPROVED** [review]

File: sdk/identity/Azure.Identity/src/Azure.Identity.csproj [hunk 3]
```diff
@@ -2,9 +2,9 @@
   <PropertyGroup>
     <Description>This is the implementation of the Azure SDK Client Library for Azure Identity</Description>
     <AssemblyTitle>Microsoft Azure.Identity Component</AssemblyTitle>
-    <Version>1.8.0</Version>
+    <Version>1.8.1</Version>
     <!--The ApiCompatVersion is managed automatically and should not generally be modified manually.-->
-    <ApiCompatVersion>1.7.0</ApiCompatVersion>
```
Comment (heaths) [inline sdk/identity/Azure.Identity/src/Azure.Identity.csproj:7]: I didn't catch this (lack of) change. It's concerning the bot didn't update this before. Was this branched from the 1.7.0 commit, can I safely assume?

**Comment by snakefoot** [comment]
Seems #32498 has [not been fixed completely](https://github.com/Azure/azure-sdk-for-net/issues/32498#issuecomment-1382669582)

**Pull request merged**